### PR TITLE
Add randomNormal.match

### PR DESCRIPTION
**Pull request by lesters**

### CODE REVIEW

File: searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp [hunk 1]
```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp]: Please update year

File: searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        _seed = util::strToNum<uint64_t>(p.get());
+    }
+    if (params.size() > 0) {
+        _mean = params[0].asDouble();
+    }
+    if (params.size() > 1) {
+        _stddev = params[1].asDouble();
+    }
+
+    describeOutput("out" , "A random value drawn from the Gaussian distribution that is stable for a given Stable (document and query)");
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp]: Typo? "... stable for a given Stable ..."

File: searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    return true;
+}
+
+search::fef::FeatureExecutor &
+RandomNormalStableBlueprint::createExecutor(const search::fef::IQueryEnvironment &env, vespalib::Stash &stash) const
+{
+    uint64_t seed = util::strToNum<uint64_t>
+            (env.getProperties().lookup(getName(), "seed").get("1024")); // default seed
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp]: Shouldn't we be using _seed if "seed" override in query is not set?

Comment (geirst) [reply searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp]: If so, _seed should be initialized to 1024 if that is the default

File: searchlib/src/vespa/searchlib/util/random_normal.h [hunk 4]
```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/util/random_normal.h]: Please update year

File: searchlib/src/vespa/searchlib/util/random_normal.h [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+#pragma once
+
+#include <vespa/searchlib/util/rand48.h>
+#include <cmath>
+
+namespace search {
+
+class RandomNormal
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/util/random_normal.h]: Please add class comment

File: searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+#include <vespa/log/log.h>
+LOG_SETUP(".features.randomnormalstablefeature");
+
+namespace search {
+namespace features {
+
+RandomNormalStableExecutor::RandomNormalStableExecutor(uint64_t seed, double mean, double stddev) :
+    search::fef::FeatureExecutor(),
+    _rnd(mean, stddev, true),
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/random_normal_stable_feature.cpp]: Shouldn't useSpare be false? We are re-seeding on each call to execute and need a new calculation when calling next() on RandomNormal.

File: searchlib/src/vespa/searchlib/features/random_normal_stable_feature.h [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    uint64_t     _seed;
+
+public:
+    RandomNormalStableExecutor(uint64_t seed, double mean, double stddev);
+    void execute(uint32_t docId) override;
+};
+
+
+/**
+ * Implements the blueprint for the random normal feature.
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/features/random_normal_stable_feature.h]: Please add 'stable' to description

**Review by geirst — CHANGES_REQUESTED** [review]
Overall looks good, please have a look at my comments

**Comment by lesters** [comment]
@geirst Thank you, comments have been addressed.

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**